feature/sqlite-browser

kvithayathil opened the pull request:
Still somewhat of a WIP (apologies for the large PR) but this module does the following:

- Show list of databases files in the app's local database folder
- Explore tables within that database
- View records within a selected table (with a 2-way scroll view).


What still needs to be done/nice to have
- Pagination on viewing records. Right now it loads the entire table in to memory, so large tables will become an issue.
- Show table/column schemas in the table viewer
- Dynamically update if CUD operations are performed
- Allow for queries to be executed
- Ability to load/view database files from outside the app
- Probably a better sample dataset

kvithayathil committed:
- Adding initial module files
- Create a database with pre-populated entries in the sample app

kvithayathil committed:
Adding Sqlite module selection to drawer

kvithayathil committed:
Display list of databases in database list activity

kvithayathil committed:
WIP Table list viewer

- Placeholder Activity to display a list of tables in a selected database (WIP)
- Refactored package name

kvithayathil committed:
- Re-organising of dependencies/versioning for sample and plugin
- Refactoring database list activity to filter out journal databases from list

kvithayathil committed:
- WIP Listing tables in selected database
- Using viewmodels and rxjava for asynchrnous query and display of results

kvithayathil committed:
- Sample data updated to demonstrate scrolling and blob filtering
- Adding screen for viewing the database records

ToxicBakery commented on hyperion-sqlite/src/androidTest/java/com/willowtreeapps/hyperion/sqlite/ExampleInstrumentedTest.java:
```diff
@@ -0,0 +1,26 @@
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
```
Probably best not to leave sample code in

ToxicBakery commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/presentation/database/DatabaseListAdapter.java:
```diff
@@ -0,0 +1,64 @@
+
+        @Override
+        public void onClick(View v) {
+            TablesListActivity.startActivity(v.getContext(), databaseName);
```
This should really be an interface to prevent tightly coupling the activity to the adapter.

ToxicBakery commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/presentation/tables/TablesListAdapter.java:
```diff
@@ -0,0 +1,72 @@
+
+        @Override
+        public void onClick(View v) {
+            DbRecordViewerActivity.startActivity(v.getContext(), databaseName,
```
Tight coupling

ToxicBakery commented on hyperion-sqlite/src/test/java/com/willowtreeapps/hyperion/sqlite/ExampleUnitTest.java:
```diff
@@ -0,0 +1,17 @@
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class ExampleUnitTest {
```
Sample class

ToxicBakery reviewed (commented):
:+1: nice work!

kvithayathil committed:
- Removing sample tests

Kritarie commented on hyperion-sqlite/src/main/res/layout/hsql_database_file_viewholder.xml:
```diff
@@ -0,0 +1,21 @@
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        tools:text="sample_db"
+        />
```
should put ending tag on same line

Kritarie commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/domain/usecase/BaseUseCase.java:
```diff
@@ -0,0 +1,36 @@
+        return this.db;
+    }
+
+    protected Single<Cursor> executeRawQuery(final String query, final String[] selectionArgs) {
```
thinking this should be final?

Kritarie commented on hyperion-sqlite/proguard-rules.pro:
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
+# You can control the set of applied configuration files using the
+# proguardFiles setting in build.gradle.
```
can delete this auto-generated

Kritarie commented on hyperion-sample/src/main/java/com/willowtreeapps/hyperion/sample/database/SampleDatabase.java:
```diff
@@ -0,0 +1,67 @@
+    private static SampleDatabase INSTANCE;
+
+    public synchronized static SampleDatabase getInstance(@NonNull final Context context) {
+        if(INSTANCE == null) {
```
should have space after `if`

Kritarie commented on hyperion-sqlite/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,11 @@
+    package="com.willowtreeapps.hyperion.sqlite">
+    <application>
+        <activity android:name=".presentation.database.DatabaseListActivity"
+            />
```
prefer same lane ending tags here

Kritarie commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/presentation/database/DatabaseListActivity.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.List;
+import java.util.Set;
+
+public class DatabaseListActivity extends AppCompatActivity {
```
should add `@HyperionIgnore` annotation to this Activity, so the menu is not embedded into Plugin activities

Kritarie commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/presentation/records/DbRecordViewerActivity.java:
```diff
@@ -0,0 +1,63 @@
+
+import io.reactivex.functions.Consumer;
+
+public class DbRecordViewerActivity extends AppCompatActivity {
```
`@HyperionIgnore` here as well

Kritarie commented on hyperion-sqlite/src/main/java/com/willowtreeapps/hyperion/sqlite/presentation/tables/TablesListActivity.java:
```diff
@@ -0,0 +1,68 @@
+
+import io.reactivex.functions.Consumer;
+
+public class TablesListActivity extends AppCompatActivity {
```
`@HyperionIgnore`

Kritarie commented on hyperion-sample/build.gradle:
```diff
@@ -35,6 +35,9 @@ dependencies {
     implementation "com.android.support:recyclerview-v7:${project.versionSupportLibrary}"
     implementation "com.jakewharton.timber:timber:4.7.0"
+    implementation "android.arch.persistence.room:runtime:1.1.0"
+    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
```
I may have missed something; why does the sample app need constraint layout?

ToxicBakery replied:
I believe constraint layout is added by default now so this would be the same issue as the auto-generated tests.

Kritarie reviewed (commented):
A few small things, but otherwise good stuff! :+1:

kvithayathil committed:
- Removing auto-generated proguard
- Removing constraint-layout dependency
- Formatting updates
- Adding @HyperionIgnore so the menu is not embedded into plugin's own activities

kvithayathil committed:
- Null-check on weak reference prior to closing the database
- Removed redundant constraint layout dependency

kvithayathil committed:
- Using listeners to handle click events on list items for database/table selection
- Additional filtering on -wal and -shm files
